editor/src/components/canvas/canvas-strategies/strategies/reparent-utils.ts

```diff
@@ -162,3 +174,44 @@ export function cursorForMissingReparentedItems(
 
   return null
 }
+
+export function placeholderCloneCommands(
+  canvasState: InteractionCanvasState,
+  customStrategyState: CustomStrategyState,
+  filteredSelectedElements: Array<ElementPath>,
+  newParent: ElementPath,
+): { commands: Array<CanvasCommand>; duplicatedElementNewUids: { [elementPath: string]: string } } {
+  let duplicatedElementNewUids: { [elementPath: string]: string } = {
+    ...customStrategyState.duplicatedElementNewUids,
+  }
+  const commands = filteredSelectedElements.flatMap((elementPath) => {
+    const element = MetadataUtils.findElementByElementPath(
+      canvasState.startingMetadata,
+      elementPath,
+    )
+    // we want to keep a placeholder element where the original dragged element was to avoid the new parent shifting around on the screen
+    const hasCommonAncestor =
```

Comment:
I think it's worth adding a comment here that THIS is the entry point if we want a more finegrained heuristic for happy paths where we don't need the layout-keeping